utils.ts

```diff
@@ -0,0 +1,17 @@
+/** @returns Alphabetically sorted object keys */
+export function sortRecord(object: Record<string, any>) {
+  return Object.keys(object)
+    .sort()
+    .reduce((sortedRecord: Record<string, any>, key) => {
+      sortedRecord[key] = object[key];
+
+      return sortedRecord;
+    }, {});
+}
+
+/** @returns DOType/DAType/EnumType reference by the elements type attribute */
+export function referencedDataType(element: Element): Element | undefined {
+  return Array.from(element.closest("DataTypeTemplates")?.children ?? []).find(
+    (sibling) => sibling.getAttribute("id") === element.getAttribute("type")
+  )!;
```

Comment:
A small performance improvement could be got by the not much less readable

```suggestion
  const elementType = element.getAttribute("type");
  return Array.from(element.closest("DataTypeTemplates")?.children ?? []).find(
    (sibling) => sibling.getAttribute("id") === elementType
  )!;
```

Reply:
True and better readable. Thank you